src/GHC/Util.hs

```diff
@@ -345,30 +351,39 @@ pragmas anns =
     [ (c, s) |
         c@(L _ (AnnBlockComment comm)) <- fromMaybe [] $ Map.lookup noSrcSpan (snd anns)
       , let body = trimCommentDelims comm
-      , "#" `isPrefixOf` body && "#" `isSuffixOf` body
-      , let s = trim $ take (length body - 2) (drop 1 body)
+      , Just rest <- [stripSuffix "#" =<< stripPrefix "#" body]
+      , let s = trim rest
     ]
 
+-- Utility for a case insensitive prefix strip.
+stripPrefixCI :: String -> String -> Maybe String
+stripPrefixCI pref str =
+  let pref' = Text.toCaseFold (Text.pack pref)
+      (str_pref, rest) = Text.splitAt (Text.length pref') (Text.pack str)
+  in if Text.toCaseFold str_pref == pref' then Just (Text.unpack rest) else Nothing
```

Comment:
Why go via Text here? Why not just use `lower` to make things lower case? Or `upper` to make them upper case.

Reply:
https://github.com/ndmitchell/hlint/pull/736